### PR TITLE
Running updates on events and low hanging fruit

### PR DESCRIPTION
**Pull request by 0xilly**
No finished but its a good start and mostly of the easy stuff is taken care of

### CODE REVIEW

File: patches/minecraft/net/minecraft/block/Block.java.patch [hunk 1]
```diff
@@ -9,3 +9,60 @@
...
++   /* ==============================A========== FORGE START =====================================*/
++   private ThreadLocal<IBlockState> silk_check_state = new ThreadLocal();
++
++   public boolean canSilkHarvest(IWorld world, BlockPos pos, IBlockState state, EntityPlayer player)
++   {
++       silk_check_state.set(state);
++       boolean ret = this.canSilkHarvest();
++       silk_check_state.set(null);
++       return ret;
+ }
```
Comment (0xilly) [inline patches/minecraft/net/minecraft/block/Block.java.patch]: Fixed in local branch

File: src/main/java/net/minecraftforge/common/ForgeHooks.java [hunk 2]
```diff
@@ -208,7 +208,7 @@ public static ItemStack getGrassSeed(Random rand, int fortune)
     private static boolean toolInit = false;
     //static HashSet<List> toolEffectiveness = new HashSet<List>();
 
-    public static boolean canHarvestBlock(@Nonnull Block block, @Nonnull EntityPlayer player, @Nonnull IBlockAccess world, @Nonnull BlockPos pos)
+    public static boolean canHarvestBlock(@Nonnull Block block, @Nonnull EntityPlayer player, @Nonnull IWorld world, @Nonnull BlockPos pos)
```
Comment (bs2609) [inline src/main/java/net/minecraftforge/common/ForgeHooks.java:212]: Should this not use the read-only interface if it's replacing `IBlockAccess`?

**Comment by loordgek** [comment]
i heard the plan was to move to mixins and default interfaces

**Comment by 0xilly** [comment]
Mixins isn't updated yet and right now we just want forge to compile

**Comment by LexManos** [comment]
Yes, we are wanting to move to default interfaces, Mixins arnt coming anytime soon as we have STILL not solved the Dev-time issues.

**Comment by loordgek** [comment]
what for issues are you running into

**Comment by 0xilly** [comment]
There was a problem re basing moving another PR

File: src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+     * Called throughout the code as a replacement for block instanceof BlockContainer
+     * Moving this to the Block base class allows for mods that wish to extend vanilla
+     * blocks, and also want to have a tile entity on that block, may.
+     *
+     * Return true from this function to specify this block has a tile entity.
+     *
+     * @param state State of the current block
+     * @return True if block has a tile entity, false otherwise
+     */
+    default boolean hasTIleEntity(IBlockState state)
```
Comment (liach) [inline src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java:121]: typo

Comment (0xilly) [reply src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java:121]: see #5138

**Pull request closed without merging**